### PR TITLE
Tried to add BlockFailedDispenseEvent

### PR DESCRIPTION
**Pull request by LB45440078L**
This event should trigger when no item is being dispensed , but the dispenser is still being triggered.
I tried my best

### CODE REVIEW

File: Spigot-API-Patches/0199-Added-new-BlockFailedDispense-event.patch [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Called when a block tries to dispense an item, but is empty.
++ * <p>
++ * Cancelling this event will make the block not try to dispense
++ * any item.
++ *
++ * @author TheViperShow
```
Comment (kennytv) [inline Spigot-API-Patches/0199-Added-new-BlockFailedDispense-event.patch]: Don’t think author tags are needed

File: Spigot-API-Patches/0199-Added-new-BlockFailedDispense-event.patch [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
++     * @param cancel true if you wish to cancel this event
++     */
++    @Override
++    public void setCancelled(boolean cancel) {
++        cancelled = cancel;
++    }
++
++    @Nullable
++    @Override
++    public @NotNull HandlerList getHandlers() {
```
Comment (kennytv) [inline Spigot-API-Patches/0199-Added-new-BlockFailedDispense-event.patch]: The handler list is notnull (right now you have both nullable and notnull annotations 👀)

File: Spigot-API-Patches/0199-Added-new-BlockFailedDispense-event.patch [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
++        return handlers;
++    }
++
++    @Nullable
```
Comment (kennytv) [inline Spigot-API-Patches/0199-Added-new-BlockFailedDispense-event.patch]: Should be NotNull

File: Spigot-API-Patches/0199-Added-new-BlockFailedDispense-event.patch [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
++    }
++
++    /**
++     * Gets the cancellation state of this event. A cancelled event will not
++     * be executed in the server, but will still pass to other plugins
++     *
++     * @return true if this event is cancelled
++     */
++    @Override
++    public boolean isCancelled() {
```
Comment (kennytv) [inline Spigot-API-Patches/0199-Added-new-BlockFailedDispense-event.patch]: How is an event doing nothing cancellable? :D

File: Spigot-Server-Patches/0492-Added-new-BlockFailedDispenseEvent.patch [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ import java.util.List;
+ import java.util.Random;
+@@ -16,6 +17,11 @@ import org.bukkit.event.world.StructureGrowEvent;
+ public interface IDispenseBehavior {
+ 
+     IDispenseBehavior NONE = (isourceblock, itemstack) -> {
++        // Paper start - register empty item
++        World world = isourceblock.getWorld();
++        org.bukkit.block.Block bukkitBlock = world.getWorld().getBlockAt(isourceblock.getBlockPosition().getX(), isourceblock.getBlockPosition().getY(), isourceblock.getBlockPosition().getZ());
++        BlockFailedDispenseEvent blockFailedDispenseEvent = new BlockFailedDispenseEvent(bukkitBlock);
```
Comment (Trigary) [inline Spigot-Server-Patches/0492-Added-new-BlockFailedDispenseEvent.patch]: You don't do anything with the event instance. You need to call it. I think it's: `Bukkit.getPluginManager().callEvent(Event)` And if it is `Cancellable` you need to check whether it's cancelled and act based on that.

Comment (Trigary) [reply Spigot-Server-Patches/0492-Added-new-BlockFailedDispenseEvent.patch]: Aikar told me that I'm a stupid Bukkit guy who can't use Paper, so what you actually need to do is just `Event#callEvent`, which also returns whether the event is cancelled.

**Pull request closed without merging**